mira/sources/vensim.py

```diff
@@ -0,0 +1,360 @@
+                unit_text = (
+                    model_parameter_info["Units"].values[0].replace(" ", "")
+                )
+                parameter = {
+                    "id": name,
+                    "value": float(expression),
+                    "description": model_parameter_info["Comment"].values[0],
+                    "units": {"expression": unit_text},
+                }
+            else:
+                parameter = {
+                    "id": name,
+                    "value": float(expression),
+                    "description": model_parameter_info["Comment"].values[0],
+                }
+            mira_parameters[name] = parameter_to_mira(parameter)
+
+    # add initials as parameters
+    state_initial_values = model.run().iloc[0]
+    for name, param_val in state_initial_values.items():
+        py_name = old_name_new_pyname_map.get(name)
+        if py_name in concepts:
+            param_name = str(mira_initials[py_name].expression)
+            param_description = "Total {} count at timestep 0".format(py_name)
+
+            if concepts[py_name].units:
+                unit_text = str(concepts[py_name].units.expression).replace(
+                    " ", ""
+                )
+                parameter = {
+                    "id": param_name,
+                    "value": param_val,
+                    "description": param_description,
+                    "units": {"expression": unit_text},
+                }
+            else:
+                parameter = {
+                    "id": param_name,
+                    "value": param_val,
+                    "description": param_description,
+                }
+            mira_parameters[param_name] = parameter_to_mira(parameter)
+
+    # construct transitions mapping that determine inputs and outputs states to a rate-law
+    transition_map = {}
+    auxiliaries = model_doc_df[model_doc_df["Type"] == "Auxiliary"]
+    for index, aux_tuple in auxiliaries.iterrows():
+        if (
+            aux_tuple["Subtype"] == "Normal"
+            and aux_tuple["Real Name"] not in CONTROL_VARIABLE_NAMES
+        ):
+            rate_name = aux_tuple["Py Name"]
+            input, output, controller = None, None, None
+
+            # If we come across a rate-law that is leaving a state, we add the state as an input
+            # to the rate-law, vice-versa if a rate-law is going into a state.
+            for state_name, in_out in state_rate_map.items():
+                if rate_name in in_out["outputs"]:
+                    input = state_name
+                if rate_name in in_out["inputs"]:
+                    output = state_name
+                    # go through outputs to get controllers. If the expression for
+                    # determining a state has multiple rate laws associated with its expression,
+                    # classify the output as a controller
+                    state_expr_sympy = state_sympy_map[output]
```

Comment:
This logic might not be general enough. We could try to recognize controllers as terms that affect the rate of the flow and either appear both as input/output or don't appear as input/output at all.

Reply:
E.g., `infectious` will appear in the rate for the `succumbing` flow, but `succumbing` does not appear in the `outputs` list for `infectious` meaning that `infectious` is not consumed by `succumbing`. In other words, something that isn't consumed by a flow but that affects the rate of the flow is a controller.